timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -308,6 +309,107 @@ void declare_context(py::module &m) {
             Coordinates have shape (F, N, 3)
             Boxes have shape (F, 3, 3)
 
+    )pbdoc")
+        .def(
+            "multiple_steps_local_selection",
+            [](timemachine::Context &ctxt,
+               const int n_steps,
+               const int reference_idx,
+               const py::array_t<int, py::array::c_style> &selection_mask,
+               const int burn_in,
+               const int store_x_interval,
+               const double radius,
+               const double k) -> py::tuple {
+                if (n_steps <= 0) {
+                    throw std::runtime_error("local steps must be at least one");
+                }
+                if (burn_in < 0) {
+                    throw std::runtime_error("burn in steps must be greater or equal to zero");
+                }
+                // Lower bound on radius selected to be 1 Angstrom, to avoid case where no particles
+                // are moved. TBD whether or not this is a good lower bound
+                const double min_radius = 0.1;
+                if (radius < min_radius) {
+                    throw std::runtime_error("radius must be greater or equal to " + std::to_string(min_radius));
+                }
+                if (k < 1.0) {
+                    throw std::runtime_error("k must be at least one");
+                }
+                // TBD determine a more precise threshold, currently 10x what has been tested
+                const double max_k = 1000000.0;
+                if (k > max_k) {
+                    throw std::runtime_error("k must be less than than " + std::to_string(max_k));
+                }
```

Comment:
todo [ future ] -- consolidate `validate_local_md_settings()` that can be used by this function and `multiple_steps_local`

Reply:
Done in https://github.com/proteneer/timemachine/pull/1005/commits/d9d511a09a7e3a09ecb0de50ccba1b85e2704609